editors/code/src/config.ts

```diff
@@ -280,18 +287,24 @@ export class Config {
     }
 }
 
-export function substituteVSCodeVariables<T>(resp: T): T {
+export function prepareVSCodeConfig<T>(
+    resp: T,
+    cb?: (key: Extract<keyof T, string>, res: { [key: string]: any }) => void
```

Comment:
comment as to what this callback is for would be nice

Reply:
documented this!

(I'm not sure I'm happy with this approach overall, but it seems to work well for now...)